### ISSUE TITLE
Please add support for the description of Tag objects

### ISSUE THREAD

**Issue by kc284** [swagger-api/swagger-codegen#1073]
The name of a tag object is used to group all operations marked with this tag under the same file. I would like to be able to output the description of the tag object as well in this file. For example:

Swagger:

```
"tags": [{ "name": "other", "description": "my description" }]
```

C# library:

```
/// <summary>
/// my description
/// </summary
public class OtherApi { }
```

**Comment by wing328** [comment]
@kc284 I would suggest you to make the suggestion in the [`swagger-spec`](https://github.com/swagger-api/swagger-spec/issues) project as `swagger-codegen` is just one of the tools that consumes the Swagger Spec.

**Comment by kc284** [comment]
@wing328 According to the [TagObject](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#tagObject) specification the syntax is already supported, my issue is that the generator picks and uses only the name ignoring the other fields.

**Comment by tckb** [comment]
+1 the support.

**Comment by splintor** [comment]
+1
I'll be glad to have this fixed.
I'm building a JS CLI, and would like the list the tags and their descriptions. Currently, it seems I'll have to resort to hard-coding the description, but I really don't like it.

**Comment by akhilkedia94** [comment]
+1
I'll be glad to have this fixed too.
We are using Swagger Codegen to develope client SDKs, and then further automatically generating documentation for the SDKs using comment tags.
Having a description will make it significantly more clear what each API class is about.

I'm open to contributing to Swagger-codegen myself to implement this, if I can have some pointers as to how to get started to contribute - perhaps the developers have some IRC channel?

**Comment by patrickheinzelmann** [comment]
I think, it should be a simple 2-step fix.

1. Extend the model class Swagger of swagger-core with a new method to return the Tag for a tag name.
2. In the de DefaultGenerator of swagger-codegen, you have to extend the method "processOperations" to add a new placeholder (for example: "classDescription") which contains the description of the tag.

If step 1 isn't possible, you can also loop through the tags in step 2. In both cases, easy to fix.

I would fix it, but I need the recommendation to go with the 2-step solution or solve it entirely inside of the generator. Is the swagger-core accepting pull requests solving problems inside of swagger-codegen?

**Comment by wing328** [comment]
@patrickheinzelmann thanks for offering help to add this feature.

I suggest we go with step 1 first and likely swagger core will welcome such enhancement as tag description should also be represented/stored by swagger core.

**Comment by patrickheinzelmann** [comment]
@wing328 I opened an issue at swagger-core. Tag description is already represented/stored in swagger-core, the issue is about avoiding the duplication of looping structures.

**Comment by patrickheinzelmann** [comment]
@wing328 I created an issue, fixed it and submitted a pull request for step 1.

**Comment by wing328** [comment]
Closed via #5455 (in 2.3.0). Please pull the latest to give it a try.
